lib/services/presets.go

```diff
@@ -877,6 +923,72 @@ func AddRoleDefaults(role types.Role) (types.Role, error) {
 	return role, nil
 }
 
+func mergeStrings(dst, src []string) (merged []string, changed bool) {
+	items := utils.NewSet[string](dst...)
+	items.Add(src...)
+	if len(items) == len(dst) {
+		return dst, false
+	}
+	dst = items.Elements()
+	slices.Sort(dst)
+	return dst, true
+}
+
+func applyAccessRequestConditionDefaults(role types.Role, enterprise bool) bool {
+	defaults := defaultAllowAccessRequestConditions(enterprise)[role.GetName()]
+	if defaults == nil {
+		return false
+	}
+
+	target := role.GetAccessRequestConditions(types.Allow)
+	changed := false
+	if target.IsEmpty() {
+		target = *defaults
+		changed = true
+	} else {
+		var rolesUpdated bool
+
+		target.Roles, rolesUpdated = mergeStrings(target.Roles, defaults.Roles)
+		changed = changed || rolesUpdated
+
+		target.SearchAsRoles, rolesUpdated = mergeStrings(target.SearchAsRoles, defaults.SearchAsRoles)
+		changed = changed || rolesUpdated
```

Comment:
Hmm isn't this the equivalent (and cleaner) or am I missing something?

Same below.

```suggestion
		target.Roles, changed = mergeStrings(target.Roles, defaults.Roles)
		target.SearchAsRoles, changed = mergeStrings(target.SearchAsRoles, defaults.SearchAsRoles)
```

Reply:
If the first `mergeStrings` call updates the role but the second one doesn't then the `changed` flag will be incorrectly reset.